src/irmin-pack/s.ml

```diff
@@ -55,41 +55,72 @@ module type Specifics = sig
   (** [flush t] flush read-write pack on disk. Raises [RO_Not_Allowed] if called
       by a readonly instance.*)
 
-  val start_gc :
-    ?unlink:bool ->
-    throttle:[ `Block | `Skip ] ->
-    repo ->
-    commit_key ->
-    bool Lwt.t
-  (** [start_gc] tries to start the gc process and returns true if the gc is
-      launched.
+  module Gc : sig
+    (** GC *)
+
+    type throttle = [ `Block | `Skip ]
+    (** Type to control how to handle an already running GC process.
+
+        [Block] means to wait until it finishes, and iff it succeeds, start a
+        new GC process
+
+        [Skip] means ignore requests for a new GC process if one is already
+        running *)
+
+    (** {2 Low-level API} *)
+    val start_exn :
+      ?unlink:bool -> throttle:throttle -> repo -> commit_key -> bool Lwt.t
+    (** [start_exn] tries to start the gc process and returns true if the gc is
+        launched.
+
+        If [unlink] is false then temporary files and files from the previous
+        generation will be kept on disk after the gc finished. This option is
+        useful for debugging. The default is true.
 
-      If [unlink] is false then temporary files and files from the previous
-      generation will be kept on disk after the gc finished. This option is
-      useful for debugging. The default is true.
+        If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
+        false immediately without launching a second gc. If [throttle] is
+        [Block] and there is a concurrent gc, [start_gc] blocks waiting for the
+        previous gc to finish and only after launches a second gc. If the
+        previous GC failed, the function returns without launching a new GC.
```

Comment:
> If the previous GC failed, the function returns without launching a new GC.
Is it actually true? Not sure that's the actual behavior

Reply:
This comment was copied from the existing `start_gc` implementation but I can evaluate. 

I agree with your comment here and on finalise @tomjridge that we need to more closely consider how we are handling multiple calls to start/finalize.